src/main/java/com/mraof/minestuck/block/redstone/StatStorerBlock.java

```diff
@@ -0,0 +1,105 @@
+import com.mraof.minestuck.effects.CreativeShockEffect;
+import com.mraof.minestuck.tileentity.redstone.StatStorerTileEntity;
+import com.mraof.minestuck.util.ParticlesAroundSolidBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.particles.RedstoneParticleData;
+import net.minecraft.state.IntegerProperty;
+import net.minecraft.state.StateContainer;
+import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Direction;
+import net.minecraft.util.Hand;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+/**
+ * Outputs a redstone signal proportional to the recorded value its tile entity is currently set to, divided by the divide by value stored also stored in the tile entity.
+ * Stats are increased when the relevant event(in ServerEventHandler) occurs within 16 blocks of a stat storer. GUI is limited by creative shock
+ */
+public class StatStorerBlock extends Block
+{
+	public static final IntegerProperty POWER = BlockStateProperties.POWER;
+	
+	public StatStorerBlock(Properties properties)
+	{
+		super(properties);
+		this.registerDefaultState(stateDefinition.any().setValue(POWER, 0));
+	}
+	
+	@Override
+	@SuppressWarnings("deprecation")
+	public ActionResultType use(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand hand, BlockRayTraceResult hit)
+	{
+		if(worldIn.isClientSide && !player.isCrouching() && !CreativeShockEffect.doesCreativeShockLimit(player, 1))
+		{
+			TileEntity tileEntity = worldIn.getBlockEntity(pos);
+			if(tileEntity instanceof StatStorerTileEntity)
+			{
+				StatStorerTileEntity te = (StatStorerTileEntity) tileEntity;
+				
+				MSScreenFactories.displayStatStorerScreen(te);
+			}
+		}
+		
+		return ActionResultType.SUCCESS;
+	}
+	
+	@Override
+	public int getSignal(BlockState blockState, IBlockReader blockAccess, BlockPos pos, Direction side)
+	{
+		return blockState.getValue(POWER);
+	}
+	
+	@Override
+	public boolean isSignalSource(BlockState state)
+	{
+		return state.getValue(POWER) > 0;
+	}
+	
+	@Override
+	public boolean canConnectRedstone(BlockState state, IBlockReader world, BlockPos pos, @Nullable Direction side)
+	{
+		return true;
+	}
+	
+	@Override
+	public boolean hasTileEntity(BlockState state)
+	{
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	public TileEntity createTileEntity(BlockState state, IBlockReader world)
+	{
+		return new StatStorerTileEntity();
+	}
+	
+	@Override
+	public void animateTick(BlockState stateIn, World worldIn, BlockPos pos, Random rand)
+	{
+		if(stateIn.getValue(POWER) > 0)
+		{
+			if(rand.nextInt(16 - stateIn.getValue(POWER)) == 0)
```

Comment:
I'd suggest changing this similarly to how I described it for the item magnet block.